cloudknot/data/docker_reqs_ref_data/py3/ref1/unit-testing-func.py

```diff
@@ -16,34 +16,41 @@ def wrapper(*args, **kwargs):
             if array_job:
                 array_index = os.environ.get("AWS_BATCH_JOB_ARRAY_INDEX")
             else:
-                array_index = '0'
+                array_index = "0"
 
             jobid = os.environ.get("AWS_BATCH_JOB_ID")
 
             if array_job:
-                jobid = jobid.split(':')[0]
-
-            key = '/'.join([
-                'cloudknot.jobs',
-                os.environ.get("CLOUDKNOT_S3_JOBDEF_KEY"),
-                jobid,
-                array_index,
-                '{0:03d}'.format(int(os.environ.get("AWS_BATCH_JOB_ATTEMPT"))),
-                'output.pickle'
-            ])
+                jobid = jobid.split(":")[0]
+
+            key = "/".join(
+                [
+                    "cloudknot.jobs",
+                    os.environ.get("CLOUDKNOT_S3_JOBDEF_KEY"),
+                    jobid,
+                    array_index,
+                    "{0:03d}".format(int(os.environ.get("AWS_BATCH_JOB_ATTEMPT"))),
+                    "output.pickle",
+                ]
+            )
 
             result = f(*args, **kwargs)
 
             # Only pickle output and write to S3 if it is not None
             if result is not None:
-                pickled_result = cloudpickle.dumps(result)
+                pickled_result = cloudpickle.dumps(result, protocol=3)
```

Comment:
I was thinking this would be something like: 

```suggestion
                pickled_result = cloudpickle.dumps(result, protocol=cloudknot.DEFAULT_PICKLE_PROTOCOL)
```

And that this variable would be defined in the library init, so we can increment in one place in the future.

Reply:
It's like that in `dockerimage.py`. However, these are the reference files for unit testing. When cloudknot uses the script template to create the files, the unit tests then check the result against these reference files.

Whenever we update `cloudknot.DEFAULT_PICKLE_PROTOCOL` in the cloudknot library we'll have to update these reference files for the unit tests (which won't be hard to remember because the unit tests will fail).

I chose this over using `cloudknot.DEFAULT_PICKLE_PROTOCOL` in the script template itself so that we don't need cloudknot as a dependency in the generated python script.

Reply:
That makes sense!